django/db/backends/base/schema.py

```diff
@@ -1027,17 +1039,32 @@ def create_unique_name(*args, **kwargs):
         else:
             name = self.quote_name(name)
         columns = Columns(table, columns, self.quote_name)
-        return Statement(
-            self.sql_create_unique,
-            table=table,
-            name=name,
-            columns=columns,
-        )
+        if condition:
+            if not self.connection.features.supports_partial_indexes:
+                return None
+            return Statement(
+                self.sql_create_unique_index,
+                table=table,
+                name=name,
+                columns=columns,
+                condition=' WHERE ' + condition,
+            )
+        else:
+            return Statement(
+                self.sql_create_unique,
+                table=table,
+                name=name,
+                columns=columns,
+            )
 
-    def _delete_unique_sql(self, model, name):
+    def _delete_unique_sql(self, model, name, condition=None):
+        if condition:
+            if not self.connection.features.supports_partial_indexes:
+                return None
+            return self._delete_constraint_sql(self.sql_delete_index, model, name)
         return self._delete_constraint_sql(self.sql_delete_unique, model, name)
 
-    def _check_sql(self, name, check):
+    def _check_sql(self, model, name, check):
```

Comment:
Accidentally added this unused argument?

Reply:
yep, removed